src/main/java/org/terasology/simpleFarming/systems/TreeAuthoritySystem.java

```diff
@@ -0,0 +1,573 @@
+		}
+		return EntityRef.NULL;
+	}
+	
+	/**
+	 * Creates a leaf at the specified location and adds it to a tree using its root.
+	 * 
+	 * @param location The place to add the leaf block.
+	 * @param leaf The prefab for the leaf. It should have a {@link BushDefinitionComponent}
+	 * @param rootComponent The RootComponent to add the leaf's {@link EntityRef} to.
+	 * 
+	 * @return The newly created leaf's block entity.
+	 * 
+	 * @see #addLeaf(Vector3i, Prefab)
+	 * @see RootComponent#leaves
+	 */
+	private EntityRef addLeaf(Vector3i location, Prefab leaf, RootComponent rootComponent) {
+		EntityRef leafEntity = addLeaf(location, leaf);
+		if(leafEntity != null && leafEntity != EntityRef.NULL) {
+			rootComponent.leaves.add(leafEntity);
+		}
+		return leafEntity;
+	}
+	
+	/**
+	 * Destroys all of the blocks in a tree except for the root. This is done usually to make way for a new growth stage of the tree.
+	 * 
+	 * @param rootEntity The root of the tree.
+	 * @param location The root's location.
+	 */
+	private void destroyTree(EntityRef rootEntity, Vector3i location) {
+		location.addY(1);
+		EntityRef aboveLogEntity = blockEntityRegistry.getExistingEntityAt(location);
+		location.addY(-1);
+		if(aboveLogEntity != EntityRef.NULL && aboveLogEntity.hasComponent(LogComponent.class)
+				&& aboveLogEntity.getComponent(LogComponent.class).root == rootEntity) {
+			destroyLog(null, aboveLogEntity);
+		}
+	}
+
+	/**
+	 * Destroys a log block in a tree, and all of the logs above it along with all of the leaves on a tree. Also marks the tree as
+	 * no longer alive so that it does not grow anymore.
+	 * 
+	 * @param event The event for the log being destroyed. If it is present, it will be sent to all other blocks which are destroyed,
+	 * allowing them to drop items, otherwise, the blocks will simply be replaced with air, allowing for growth or un-growth.
+	 * @param log The log to be destroyed.
+	 * 
+	 * @see #destroyLeaves(Set, DoDestroyEvent)
+	 */
+	private void destroyLog(DoDestroyEvent event, EntityRef log) {
+		LogComponent logComponent = log.getComponent(LogComponent.class);
+		EntityRef rootEntity = logComponent.root;
+		RootComponent rootComponent = rootEntity.getComponent(RootComponent.class);
+		
+		if(rootComponent.alive) {
+			destroyLeaves(rootComponent.leaves, event);
+			rootComponent.leaves.clear();
+			rootComponent.alive = false;
+			rootEntity.addOrSaveComponent(rootComponent);
+		}
+		
+		if(event == null) {
+			worldProvider.setBlock(logComponent.location, airBlock);
+		}
+		
+		Vector3i aboveLocation = new Vector3i(logComponent.location).addY(1);
+		EntityRef aboveLogEntity = blockEntityRegistry.getExistingEntityAt(aboveLocation);
+		if(aboveLogEntity != EntityRef.NULL && aboveLogEntity.hasComponent(LogComponent.class)
+				&& aboveLogEntity.getComponent(LogComponent.class).root == rootEntity) {
+			if(event == null) {
+				destroyLog(null, aboveLogEntity);
+			} else {
+				aboveLogEntity.send(event);
```

Comment:
I'm not sure how I feel about sending this `DoDestroyEvent`. Admittedly I've not looked at the code for alternatives so if this is the best method/you have a justified reason for using it then that's fine by me